vidarr-server/src/test/java/ca/on/oicr/gsi/vidarr/server/PriorityByWorkflowTest.java

```diff
@@ -124,5 +124,24 @@ public void testIfWorkflowRunWIthEmptyPriorityIsSet_thenWorkflowRunIsAddedWithLo
 
   }
 
+  @Test
+  public void testIfResourceIsReleased_thenWorkflowRunIsAddedBackToWaiting() {
+
+    JsonNode higherPriority = mapper.valueToTree(4);
+    JsonNode lowerPriority = mapper.valueToTree(2);
+
+    sut.set(workflow, "qwerty", Optional.of(higherPriority));
+
+    sut.release(workflow, version, "qwerty", Optional.of(higherPriority));
```

Comment:
Normally `release` would be called after `request`, which if successful would remove the workflow run from PriorityByWorkflow - so I think the correct setup here is to call `sut.release` without calling `sut.set` and confirming that `sut.release` adds the workflow run back to the waiting set.